cli/src/maintenance.rs

```diff
@@ -945,6 +979,9 @@ pub fn try_perform_maintenance(
         .or_else(|| state.try_collect_validator_fee())
         // Same for updating the validator balance.
         .or_else(|| state.try_withdraw_inactive_stake())
+        // Try to remove validator should happen before withdrawing inactive
+        // stake and collecting validator's fees.
```

Comment:
Huh, no, it should happen after ... we can’t remove the validator if it has stake or unclaimed fees.

It doesn’t matter so much because we check if it can be removed, so it will still happen one iteration later, but I would still put this one last, because it is the least-priority thing we have to do; if we can stake a deposit, we prefer to do that sooner, rather than removing the validator.